File: 프로그래머스/스택and큐/주식가격.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> prices) {
    vector<int> answer;
    
    for(int i = 0; i < prices.size() - 1; i++){
        int number = 0;
        for(int j = i + 1; j < prices.size(); j++){
            if(prices[j] < prices[i]){
                number++;
                break;
            }
        number++;
        }
       answer.push_back(number); 
    }
    answer.push_back(0);
    return answer;
}

File: 프로그래머스_python/스택and큐/주식가격.py
def solution(prices):
    answer = [0] * len(prices)
    
    array = []
    
    i = 0
    
    while i < len(prices):

        if len(array) == 0:
            array.append(i)
            i += 1
        elif prices[array[-1]] <= prices[i]:
            array.append(i)
            i += 1
        else:
            answer[array[-1]] = i - array[-1]
            del array[-1]
    
    for i in array:
        answer[i] = len(prices) - 1 - i
    return answer
